src/algebra/field/basic.lean

```diff
@@ -50,23 +50,59 @@ set_option old_structure_cmd true
 universe u
 variables {α β K : Type*}
 
+/-- The default definition of the coercion `(↑(a : ℚ) : K)` for a division ring `K`
+is defined as `(a / b : K) = (a : K) * (b : K)⁻¹`.
+Use `coe` instead of `of_rat_rec` for better definitional behaviour.
+-/
+def of_rat_rec [has_lift_t ℕ K] [has_lift_t ℤ K] [has_mul K] [has_inv K] : ℚ → K
+| ⟨a, b, _, _⟩ := ↑a * (↑b)⁻¹
+
+/-- The default definition of the scalar multiplication `(a : ℚ) • (x : K)` for a division ring `K`
+is given by `a • x = (↑ a) * x`.
+Use `(a : ℚ) • (x : K)` instead of `qsmul` for better definitional behaviour.
+-/
+def qsmul_rec (coe : ℚ → K) [has_mul K] (a : ℚ) (x : K) : K :=
+coe a * x
+
 /-- A `division_semiring` is a `semiring` with multiplicative inverses for nonzero elements. -/
 @[protect_proj, ancestor semiring group_with_zero]
 class division_semiring (α : Type*) extends semiring α, group_with_zero α
 
-/-- A `division_ring` is a `ring` with multiplicative inverses for nonzero elements. -/
-@[protect_proj, ancestor ring division_semiring]
-class division_ring (α : Type*) extends ring α, div_inv_monoid α, nontrivial α :=
-(mul_inv_cancel : ∀ {a : α}, a ≠ 0 → a * a⁻¹ = 1)
-(inv_zero : (0 : α)⁻¹ = 0)
+/-- A `division_ring` is a `ring` with multiplicative inverses for nonzero elements.
+
+An instance of `division_ring K` includes maps `of_rat : ℚ → K` and `qsmul : ℚ → K → K`.
+If the division ring has positive characteristic p, we define `of_rat (1 / p) = 1 / 0 = 0`
+for consistency with our division by zero convention.
+The fields `of_rat` and `qsmul are needed to implement the
+`algebra_rat [division_ring K] : algebra ℚ K` instance, since we need to control the specific
+definitions for some special cases of `K` (in particular `K = ℚ` itself).
+See also Note [forgetful inheritance].
+-/
+@[protect_proj, ancestor ring div_inv_monoid nontrivial]
+class division_ring (K : Type u) extends ring K, div_inv_monoid K, nontrivial K :=
+(mul_inv_cancel : ∀ {a : K}, a ≠ 0 → a * a⁻¹ = 1)
+(inv_zero : (0 : K)⁻¹ = 0)
+(of_rat : ℚ → K := of_rat_rec)
```

Comment:
I think to match the `nat` and `int` versions, this should be extracted to a `has_rat_cast` typeclass; then `function.injective.field` should take a `[has_rat_cast F]` argument to match the existing `has_nat_cast` arguments.